[GTK] Add GTK 4 migration guide

mcatanzaro opened the pull request:
#### 99f4041e6cb2250026638cf31e650e1170af71b4
<pre>
[GTK] Add GTK 4 migration guide
<a href="https://bugs.webkit.org/show_bug.cgi?id=247133">https://bugs.webkit.org/show_bug.cgi?id=247133</a>

Reviewed by Carlos Garcia Campos.

* Source/WebKit/gtk/migrating-to-webkitgtk-6.0.md: Added.
* Source/WebKit/gtk/webkitgtk.toml.in:

Canonical link: <a href="https://commits.webkit.org/256100@main">https://commits.webkit.org/256100@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3fdcb6254893d0f0e4fb7c153b9202dc561cbd3c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/94518 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/3696 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/27423 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/104171 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/3748 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/31892 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/86846 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/100142 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/100188 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-16-Simulator-WK2-Tests-EWS "Waiting to run tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/80909 "Built successfully") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/29722 "Found 1 new test failure: media/video-seek-to-current-time.html (failure)") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/84184 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/38293 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/18008 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/36157 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/19281 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4210 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/40056 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/41926 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/42009 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/38511 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

TingPing reviewed (approved):
:+1:

carlosgcampos reviewed (approved):
We should also document that ondemand hardware acceleration setting is not available too, but we can do it in follow up patch if you prefer.

mcatanzaro committed:
[GTK] Add GTK 4 migration guide
https://bugs.webkit.org/show_bug.cgi?id=247133

Reviewed by Carlos Garcia Campos.

* Source/WebKit/gtk/migrating-to-webkitgtk-6.0.md: Added.
* Source/WebKit/gtk/webkitgtk.toml.in:

Canonical link: https://commits.webkit.org/256100@main
